pandas/core/frame.py

```diff
@@ -84,7 +84,7 @@
     find_common_type,
     infer_dtype_from_scalar,
     invalidate_string_dtypes,
-    maybe_box_datetimelike,
+    maybe_box_native,
```

Comment:
FTR the `DataFrame` tests still pass without the changes in `frame.py` so could revert them (but I think `maybe_box_native` is the right one here and maybe there are tests cases I haven't looked at)